### PR TITLE
Openni Kinect IR generator support

### PR DESCRIPTION
**Pull request by soyersoyer**

### CODE REVIEW

File: modules/videoio/include/opencv2/videoio.hpp [hunk 1]
```diff
@@ -79,7 +79,8 @@ enum { CAP_ANY          = 0,     // autodetect
        CAP_DSHOW        = 700,   // DirectShow (via videoInput)
        CAP_PVAPI        = 800,   // PvAPI, Prosilica GigE SDK
        CAP_OPENNI       = 900,   // OpenNI (for Kinect)
-       CAP_OPENNI_ASUS  = 910,   // OpenNI (for Asus Xtion)
+       CAP_OPENNI_IR    = 910,   // OpenNI (for Kinect IR mode)
```
Comment (vpisarev) [inline modules/videoio/include/opencv2/videoio.hpp]: it's actually a bad idea to change numerical values of existing enumeration constants; please, use CAP_OPENNI_IR=920 value instead

File: modules/videoio/src/cap_openni.cpp [hunk 2]
```diff
@@ -429,7 +400,7 @@ class ApproximateSyncGrabber
 class CvCapture_OpenNI : public CvCapture
 {
 public:
-    enum { DEVICE_DEFAULT=0, DEVICE_MS_KINECT=0, DEVICE_ASUS_XTION=1, DEVICE_MAX=1 };
+    enum { DEVICE_DEFAULT=0, DEVICE_MS_KINECT=0, DEVICE_MS_KINECT_IR=1, DEVICE_ASUS_XTION=2, DEVICE_MAX=2 };
```
Comment (vpisarev) [inline modules/videoio/src/cap_openni.cpp]: same thing; please, try to keep existing enumeration items unchanged.

File: modules/videoio/src/cap_openni.cpp [hunk 3]
```diff
@@ -1113,6 +1090,102 @@ bool CvCapture_OpenNI::setImageGeneratorProperty( int propIdx, double propValue
     return isSet;
 }
 
+double CvCapture_OpenNI::getIRGeneratorProperty( int propIdx ) const
+{
+    printf("getirgenprop\n");
```
Comment (vpisarev) [inline modules/videoio/src/cap_openni.cpp]: debugging printf's should be removed

**Comment by vpisarev** [comment]
@soyersoyer, thank you for the contribution! please, fix the issues mentioned above

**Comment by soyersoyer** [comment]
fixed.

**Comment by vpisarev** [comment]
@soyersoyer, thanks! can you also provide some explanation/rationale behind the changes; for example, I see that you removed embedded XML config; there is also different generator initialization. We have no ability right now to check compatibility of the new code with different PrimeSense-based devices, so we'd rather take the conservative side and accept only those changes that do not break anything

**Comment by soyersoyer** [comment]
If I want to configure the kinect in IR mode, I should have another XML config, so I decided to remove it and move the configuration to the code. The XML configuration ability is a convenience thing, so you can configure the context with code too. I assumed that the depth and the image generator works in the CAP_OPENNI  and the CAP_OPENNI_ASUS mode, but I only have a kinect, so I can't test the latter.
According to
https://github.com/Itseez/opencv/blob/master/modules/videoio/src/cap_openni.cpp#L983
the imagegenerator's initialization should be optional, so I will fix it.
I was confused because this line:
https://github.com/Itseez/opencv/blob/master/modules/videoio/src/cap_openni.cpp#L656

**Pull request closed without merging**